sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -651,7 +750,31 @@ def get_media_settings_value(physical_port, key):
     return {}
 
 
-def get_media_settings_key(physical_port, transceiver_dict):
+def get_speed_and_lane_count(port, cfg_port_tbl):
+    port_speed, lane_count = -1, -1
+    found, port_info = cfg_port_tbl.get(port)
+    if found and 'speed' in dict(port_info) and 'lanes' in dict(port_info):
+        port_speed = dict(port_info).get('speed', '-1')
+        lanes = dict(port_info).get('lanes', '-1')
+        lane_count = len(lanes.split(','))
+    return port_speed, lane_count
+
+
+def get_lane_speed_key(physical_port, port_speed, lane_count):
+    sfp = platform_chassis.get_sfp(physical_port)
+    api = sfp.get_xcvr_api()
+    speed_index = get_cmis_application_desired(api, int(lane_count), int(port_speed))
+    
+    appl_adv_dict, lane_speed_key = None, None
+    if type(api) == CmisApi:
+        appl_adv_dict = api.get_application_advertisement()
+        if speed_index is not None:
+            lane_speed_key = LANE_SPEED_KEY_PREFIX + (appl_adv_dict[speed_index].get('host_electrical_interface_id')).split()[0]
```

Comment:
is it possible that the get call would return None?

Reply:
The 'host_electrical_interface_id' key is searched for in a dictionary returned from the EEPROM. Therefore, it is highly unlikely for this key not to be present in appl_adv_dict[speed_index] (returned from get_application_advertisement()).

Just to be on the safe side, I added some protection in case an invalid dictionary was returned from the EEPROM.